src/OrchardCore/OrchardCore.Localization.Abstractions/CultureDictionaryRecordKey.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+
+namespace OrchardCore.Localization
+{
+    /// <summary>
+    /// Represents a key for <see cref="CultureDictionaryRecord"/>.
+    /// </summary>
+    public struct CultureDictionaryRecordKey : IEquatable<CultureDictionaryRecordKey>
+    {
+        private readonly string _messageId;
+        private readonly string _context;
+        private readonly string _key;
+
+        /// <summary>
+        /// Creates new instance of <see cref="CultureDictionaryRecordKey"/>.
+        /// </summary>
+        /// <param name="messageId">The message Id.</param>
+        public CultureDictionaryRecordKey(string messageId) : this(messageId, null)
+        {
+
+        }
+
+        /// <summary>
+        /// Creates new instance of <see cref="CultureDictionaryRecordKey"/>.
+        /// </summary>
+        /// <param name="messageId">The message Id.</param>
+        /// <param name="context">The message context.</param>
+        public CultureDictionaryRecordKey(string messageId, string context)
+        {
+            _messageId = messageId;
+            _context = context;
+
+            if (String.IsNullOrEmpty(context))
+            {
+                _key = messageId;
+            }
+
+            _key = String.IsNullOrEmpty(context) ? messageId : context.ToLowerInvariant() + "|" + messageId;
```

Comment:
Here you know that context is not null and not empty

Reply:
There's another constructor that sets the context to `null` ;)

Reply:
Okay i don't know the context, just because i saw the check you already did line 33, and now just saw that you always go to line 38. So, maybe lines 33 to 35 are useless or use an else block without checking again context

Reply:
No no, you are right, I just forgot to remove that check, it was from the previous code, but I change it to use ternary condition

Thanks

Reply:
Cool, at this time of the night i'm sure of nothing ;)

Reply:
Frankly it was a morning here, but I didn't sleep, so that's why I miss such thing ;)